platform/api/aws/ami.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type releaseAMIs struct {
+	AMIs []struct {
+		Name string `json:"name"`
+		PV   string `json:"pv"`
+		HVM  string `json:"hvm"`
+	}
+}
+
+func (a *API) AMI() (resolvedAMI string, resolveAMIErr error) {
+	a.opts.resolvedAMILock.Lock()
+	defer a.opts.resolvedAMILock.Unlock()
+
+	// Use cache if we can
+	if a.opts.ResolvedAMI != "" {
+		return a.opts.ResolvedAMI, nil
+	}
+
+	// If successfully resolved, cache the AMI id for next call.
+	// Decide based on the named returns
+	defer func() {
+		if resolveAMIErr == nil {
+			a.opts.ResolvedAMI = resolvedAMI
+		}
+	}()
+
+	switch a.opts.AMI {
+	case "alpha", "beta", "stable":
+	default:
+		return a.opts.AMI, nil
+	}
+
+	// Lookup the current latest AMI id via our release bucket.
+	// Note, querying the AWS API is an alternate option, but is slower and
+	// requires filtering carefully to avoid shared-but-not-public images. It
+	// also won't work as trivially for china/gov.
+	resp, err := http.DefaultClient.Get(fmt.Sprintf("https://%s.release.core-os.net/amd64-usr/current/coreos_production_ami_all.json", a.opts.AMI))
```

Comment:
Should this be using the GS code?  There doesn't seem to be much advantage in this case.